natpmp/client.go

```diff
@@ -0,0 +1,204 @@
+package natpmp
+
+import (
+	"context"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"sync"
+	"time"
+)
+
+// Version is the expected protocol version for NAT-PMP.
+const Version = 0
+
+// ErrProtocol indicates that a NAT gateway returned a response that violates
+// the NAT-PMP protocol.
+var ErrProtocol = errors.New("natpmp: protocol error")
+
+//go:generate stringer -type=Error -output=strings.go
+
+// An Error is a NAT-PMP protocol result code which indicates that an operation
+// has failed.
+type Error int
+
+// Possible Errors as defined in RFC 6886, section 3.5.
+const (
+	UnsupportedVersion Error = 1
```

Comment:
I see what you mean but I hesitate doing this for constants that I'm pulling from an RFC. I tend to only like using the implicit numbering of iota for enums I am defining in my own code.

Reply:
Fair 'nuf. I'm not super fussed either way because I don't expect people to look at the numbers once I've defined helpful constants, but your reasoning makes sense.